### PR TITLE
Add Compat.at-error, Compat.at-warn, Compat.at-info and Compat.at-debug

### PR DESCRIPTION
**Pull request by fredrikekre**
replacement for #453 which I managed to mess up....

### CODE REVIEW

**Comment by fredrikekre** [comment]
Added a reference to MicroLogging.

File: src/Compat.jl [hunk 1]
```diff
@@ -1229,6 +1229,50 @@ end
     end
 end
 
+@static if !isdefined(Base, Symbol("@info"))
+    macro info(msg, args...)
+        return :(info($(esc(msg)), prefix = "Info: "))
```
Comment (c42f) [inline src/Compat.jl]: If you want 0.6 equivalent formatting, perhaps skip the `prefix` for `@info` and `@warn`?

File: src/Compat.jl [hunk 2]
```diff
@@ -1229,6 +1229,50 @@ end
...
+    macro debug(msg, args...)
+        return :(debug($(esc(msg))))
+    end
+else
+    @eval const $(Symbol("@debug")) = Base.$(Symbol("@debug"))
+end
+@static if !isdefined(Base, Symbol("@error"))
+    function _error(msg)
+        buf = IOBuffer()
+        iob = Base.redirect(IOContext(buf, STDERR), Base.log_info_to, :error)
```
Comment (c42f) [inline src/Compat.jl]: Use `Base.log_error_to`, perhaps?

File: README.md [hunk 3]
```diff
@@ -301,6 +301,15 @@ Currently, the `@compat` macro supports the following syntaxes:
...
 * `@nospecialize` has been added ([#22666]).
 
+* The logging macros `@error`, `@warn`, `@info` and `@debug` can be used as
+  `Compat.@error`, `Compat.@warn`, `Compat.@info` and `Compat.@debug` on Julia 0.6 ([#24490]).
+  Note that the behavior do not mirror the logging macros in Julia 0.7, instead on Julia 0.6:
+  - Messages are printed to `STDERR` (like `info` and `warn` on Julia 0.6) and not to a dedicated logging stream
+  - The loglevel can not be controlled
+  - Extra metadata sent to the macros are ignored
+
+  As an alternative, see the MicroLogging.jl package for a logging interface similar to the one in Julia 0.7.
```
Comment (c42f) [inline README.md]: This is accurate. While I plan to update `MicroLogging`, it's not possible to provide exact equivalence in a package.

File: src/Compat.jl [hunk 4]
```diff
@@ -1229,6 +1229,50 @@ end
...
+    function debug(msg)
+        buf = IOBuffer()
+        iob = Base.redirect(IOContext(buf, STDERR), Base.log_info_to, :debug)
+        print_with_color(:blue, iob, "Debug: "; bold = true)
+        Base.println_with_color(:blue, iob, chomp(string(msg)))
+        print(STDERR, String(take!(buf)))
+        return
+    end
+    macro debug(msg, args...)
+        return :(debug($(esc(msg))))
```
Comment (c42f) [inline src/Compat.jl]: Debug messages are invisible by default in 0.7, so I think having them visible here by default might end up being hopelessly verbose.

Maybe a simplistic global flag which disables debug by default might be a good idea?

File: src/Compat.jl (hunk 1 above)
Comment (fredrikekre) [reply src/Compat.jl]: I did it like this to be consistent with 0.7, and also to differentiate logging errors from actual errors, which prints `ERROR: ` in all caps red.

File: src/Compat.jl (hunk 4 above)
Comment (fredrikekre) [reply src/Compat.jl]: Good idea, we can have `const DEBUG = Ref(false)` in Compat, such that people can turn it on.

**Pull request merged**